cmd/xl-storage-disk-id-check.go

```diff
@@ -1030,8 +1034,9 @@ func (p *xlStorageDiskIDCheck) checkHealth(ctx context.Context) (err error) {
 	// If also more than 15 seconds since last success, take disk offline.
 	t = time.Since(time.Unix(0, atomic.LoadInt64(&p.health.lastSuccess)))
 	if t > maxTimeSinceLastSuccess {
-		if atomic.CompareAndSwapInt32(&p.health.status, diskHealthOK, diskHealthFaulty) {
+		if p.health.status.CompareAndSwap(diskHealthOK, diskHealthFaulty) {
 			logger.LogAlwaysIf(ctx, fmt.Errorf("node(%s): taking drive %s offline, time since last response %v", globalLocalNodeName, p.storage.String(), t.Round(time.Millisecond)))
+			p.health.waiting.Add(1)
```

Comment:
It is to measure the disk is waiting and error is disk faulty.

Indicating that disk has I/O errors due to blocked/waiting I/O.

Without that we can't tell if faulty disk is due to hung syscall or not.

Reply:
Isn't this captured by the state being offline?  It is just the mixing of "offline disks" and "requests waiting for IO" I'd like to avoid.

Otherwise we should probably add a separate counter for "offline disks".

Reply:
I can add a separate counter but I thought this is good enough since waiting() will have always a non-zero value at the minimum 1 even if there are not user calls being blocked.

The I/O that is blocked is the I/O that is being performed by the drive monitoring routine.

Reply:
We also need to let them know that there are hung syscalls on the disk that can only be cleared after an OS reboot. 

So, it is also an additional observability requirement per drive.